app/src/main/java/app/grapheneos/pdfviewer/fragment/PasswordPromptFragment.kt

```diff
@@ -0,0 +1,74 @@
+package app.grapheneos.pdfviewer.fragment
+
+import android.app.Dialog
+import android.content.DialogInterface
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextUtils
+import android.text.TextWatcher
+import android.view.LayoutInflater
+import android.view.WindowManager
+import android.widget.Button
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.DialogFragment
+import app.grapheneos.pdfviewer.PdfViewer
+import app.grapheneos.pdfviewer.R
+import app.grapheneos.pdfviewer.databinding.PasswordDialogFragmentBinding
+import com.google.android.material.textfield.TextInputEditText
+
+class PasswordPromptFragment : DialogFragment() {
+
+    private lateinit var passwordDialogFragmentBinding: PasswordDialogFragmentBinding
+    private lateinit var passwordEditText : TextInputEditText
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val passwordPrompt = AlertDialog.Builder(requireContext())
+        passwordDialogFragmentBinding =
+            PasswordDialogFragmentBinding.inflate(LayoutInflater.from(requireContext()))
+        passwordEditText = passwordDialogFragmentBinding.pdfPasswordEdittext
+        passwordPrompt.setView(passwordDialogFragmentBinding.root)
+        passwordEditText.addTextChangedListener(object : TextWatcher {
+            override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}
+            override fun onTextChanged(input: CharSequence, i: Int, i1: Int, i2: Int) {
+                val positiveButton: Button =
```

Comment:
Extract this code into a separate function and call it from here and from `onStart()` to avoid code duplication.